tests/TestHarness/queries.py

```diff
@@ -708,11 +708,15 @@ def getBlockProducerByNum(self, blockNum, timeout=None, waitForBlock=True, exitO
         if waitForBlock:
             self.waitForBlock(blockNum, timeout=timeout, blockType=BlockType.head)
         block=self.getBlock(blockNum, exitOnError=exitOnError)
+        if block is None and not exitOnError:
```

Comment:
Why `and not exitOnError`?  do you want to continue into If `getBlockAttribute()`  is None in any case, `value=block[key]` in `getBlockAttribute()` will abort. I think you need to have two levels of `if`. `if block is None` and then `if not exitOnError`.

Reply:
That would provide a better error message. As is it will say you can't index into None.